src/FluentValidation/ValidatorOptions.cs

```diff
@@ -29,16 +29,124 @@ namespace FluentValidation {
+		/// </para>
+		/// </summary>
+		[Obsolete($"Use {nameof(DefaultClassLevelCascadeMode)} and/or {nameof(DefaultRuleLevelCascadeMode)} instead. " +
+			"CascadeMode will be removed in a future release. " +
+			"For more details, see https://docs.fluentvalidation.net/en/latest/conditions.html#setting-the-cascade-mode")]
+		public CascadeMode CascadeMode {
+#pragma warning disable 618
+			get {
+				if (_defaultClassLevelCascadeMode == _defaultRuleLevelCascadeMode) {
+					return _defaultClassLevelCascadeMode;
+				}
+				else if (_defaultClassLevelCascadeMode == CascadeMode.Continue && _defaultRuleLevelCascadeMode == CascadeMode.Stop) {
+					return CascadeMode.StopOnFirstFailure;
+				}
+				else {
+					throw new Exception(
+						$"There is no conversion to a single {nameof(CascadeMode)} value from the current combination of " +
+							$"{nameof(DefaultClassLevelCascadeMode)} and {nameof(DefaultRuleLevelCascadeMode)}. " +
+							$"Please use these properties instead of the deprecated {nameof(CascadeMode)} going forward.");
+				}
+			}
+
+			set {
+				DefaultClassLevelCascadeMode = value == CascadeMode.StopOnFirstFailure
+					? CascadeMode.Continue
+					: value;
+
+				DefaultRuleLevelCascadeMode = value == CascadeMode.StopOnFirstFailure
+					? CascadeMode.Stop
+					: value;
+#pragma warning restore 618
+			}
+		}
+
 		/// <summary>
-		/// Default cascade mode
+		/// <para>
+		/// Sets the default value for <see cref="AbstractValidator.ClassLevelCascadeMode"/>.
+		/// Defaults to <see cref="CascadeMode.Continue"/> if not set.
+		/// </para>
+		/// <para>
+		/// Do not set this to <see cref="CascadeMode.StopOnFirstFailure"/>.
+		/// <see cref="CascadeMode.StopOnFirstFailure"/> will be removed in a future release
+		/// and will throw an exception if used here.
+		/// </para>
 		/// </summary>
-		public CascadeMode CascadeMode { get; set; } = CascadeMode.Continue;
+		public CascadeMode DefaultClassLevelCascadeMode {
+			get => _defaultClassLevelCascadeMode;
+			set {
+#pragma warning disable 618
+				if (value == CascadeMode.StopOnFirstFailure) {
+#pragma warning restore 618
+					throw new ArgumentException(_stopOnFirstFailureArgumentMessage);
+				}
+
+				_defaultClassLevelCascadeMode = value;
+			}
+		}
+
+		/// <summary>
+		/// <para>
+		/// Sets the default value for <see cref="AbstractValidator.RuleLevelCascadeMode"/>
+		/// Defaults to <see cref="CascadeMode.Continue"/> if not set.
+		/// </para>
+		/// <para>
+		/// Do not set this to <see cref="CascadeMode.StopOnFirstFailure"/>.
+		/// <see cref="CascadeMode.StopOnFirstFailure"/> will be removed in a future release
+		/// and will throw an exception if used here.
+		/// </para>
+		/// </summary>
+		public CascadeMode DefaultRuleLevelCascadeMode {
+			get => _defaultRuleLevelCascadeMode;
+			set {
+#pragma warning disable 618
+				if (value == CascadeMode.StopOnFirstFailure) {
+#pragma warning restore 618
+					throw new ArgumentException(_stopOnFirstFailureArgumentMessage);

```

Comment:
As above, please could this be changed to a conversion to do a conversion to `Stop`